conan/internal/upload_metadata.py

```diff
@@ -18,7 +18,21 @@ def _metadata_files(folder, metadata):
     return result
 
 
-def gather_metadata(package_list, cache, metadata):
+def gather_metadata(package_list, cache, metadata: list):
+    """
+    List and configure the metadata files to be uploaded.
+    The metadata supports patterns, so it can be used to upload only a subset of the metadata files.
+    When metadata is not specified, all metadata files are uploaded.
+    But, when metadata is empty string (""), it means that no metadata files should be uploaded.
+
+    :param package_list: Package to be uploaded
+    :param cache: Conan client cache
+    :param metadata: A list of patterns of metadata that should be uploaded. Default None
+    """
+
+    if metadata == ['']:
```

Comment:
What if some user introduces ``--metadata="" --metadata="logs*"``? Maybe we want to discard it when it is being used for comparison?

Reply:
Good question, should `--metadata="" --metadata="logs*"` be considered an invalid case, and raise an error? Like:

```
if metadata and '' in metadata:
    raise ConanException("Mixing explicit empty metadata with metadata folder is invalid.")
```

Reply:
Yes, but in any case, it is better if the input data validation is done in the entry point of the API, in this case, the ``prepare(self, package_list, enabled_remotes, metadata=None)`` method in ``upload.py``, so the full ``gather_metadata()`` call could be completely skipped.

Reply:
Agreed!

Reply:
Done.